src/plugin/Manager.cc

```diff
@@ -58,6 +58,10 @@ void Manager::SearchDynamicPlugins(const std::string& dir)
 	if ( dir.empty() )
 		return;
 
+	// Do not search dot directories.
+	if ( auto p = zeek::filesystem::path(dir); p.has_filename() && p.filename().c_str()[0] == '.' )
```

Comment:
Would move this down to line 142:

https://github.com/zeek/zeek/pull/3329/files#diff-b4ae5cd2712f7a5ffe288277de92b8b111e64b71219b542713d1f525dc5c76bbR151


Would also fix the test that does `ZEEK_PLUGIN_DIR=.`. Additionally, if someone does `ZEEK_PLUGIN_DIR=.tmp/just-do/.not-ask` we should let them.

Reply:
Thanks for the review, I also added a test validating that behavior.